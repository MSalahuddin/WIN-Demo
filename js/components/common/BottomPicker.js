import React from 'react';
import PropTypes from 'prop-types';
import { Picker } from 'react-native';

const renderOptions = options =>
  options.map(({ label, value }, index) => <Picker.Item key={index} label={label} value={value} />);

const BottomPicker = ({ options, onValueChange, selectedValue }) => (
  <Picker onValueChange={onValueChange} selectedValue={selectedValue} testID="bottom-picker-container">
    {renderOptions(options)}
  </Picker>
);

BottomPicker.propTypes = {
  options: PropTypes.arrayOf(PropTypes.shape({ label: PropTypes.string.isRequired, value: PropTypes.any.isRequired }))
    .isRequired,
  selectedValue: PropTypes.string.isRequired,
  onValueChange: PropTypes.func.isRequired
};

export default BottomPicker;
