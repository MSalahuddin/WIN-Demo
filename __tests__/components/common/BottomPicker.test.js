import React from 'react';
import { render } from 'react-native-testing-library';

import BottomPicker from '../../../js/components/common/BottomPicker';

describe('Component BottomPicker', () => {
  const onPress = jest.fn();
  const options = [
    { label: 'foo', value: 'foo' },
    { label: 'bar', value: 'bar' }
  ];
  test('renders correctly', () => {
    const output = render(<BottomPicker onValueChange={onPress} options={options} selectedValue="" />);
    expect(output).toMatchSnapshot();
  });
});
