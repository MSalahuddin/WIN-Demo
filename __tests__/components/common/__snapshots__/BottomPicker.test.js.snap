// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component BottomPicker renders correctly 1`] = `
<View>
  <RCTPicker
    items={
      Array [
        Object {
          "label": "foo",
          "textColor": undefined,
          "value": "foo",
        },
        Object {
          "label": "bar",
          "textColor": undefined,
          "value": "bar",
        },
      ]
    }
    onChange={[Function]}
    selectedIndex={0}
    style={
      Array [
        Object {
          "height": 216,
        },
        undefined,
      ]
    }
    testID="bottom-picker-container"
  />
</View>
`;
